Makefile

```diff
@@ -30,6 +30,21 @@ setup:
 	GOPATH=$(CURDIR)/_tools go install github.com/twitchtv/retool/...
 	$(RETOOL) build
 
+# Make commands for twirp docs
+setup_docs:
+	echo "Ruby >= 2.3.1 must be installed"
+	gem install bundler
+	bundler install --gemfile docs/Gemfile
+
+dev_docs: setup_docs
+	cd docs && bundle exec middleman server
+
+build_docs: setup_docs
+	cd docs && bundle exec middleman build --clean
+
+publish_docs: build_docs
+	git add docs/build && git push origin `git subtree split --prefix docs/build`:gh-pages --force
```

Comment:
I want to get 👀 on this command. It builds the html files, then automatically pushes them to the gh-pages branch of whichever fork/repo you are working on. Is there a better way of doing this (a way to review the outgoing changes to the docs/build on github before they are pushed to `gh-pages`)?

Reply:
The `docs/build` folder will just hold the built HTML documents, we can still review the source, right?

It would be nice if we could push a preview, yes. I don't know that GH pages has the flexibility to allow it, though. This might be the best we'll get within github pages.